mongo/mongo.go

```diff
@@ -435,26 +441,38 @@ func EnsureServer(args EnsureServerParams) (Version, error) {
 
 func ensureServer(args EnsureServerParams, mongoKernelTweaks map[string]string) (Version, error) {
 	var zeroVersion Version
+	var err error
 	tweakSysctlForMongo(mongoKernelTweaks)
 	logger.Infof(
 		"Ensuring mongo server is running; data directory %s; port %d",
 		args.DataDir, args.StatePort,
 	)
 
+	if featureflag.Enabled(feature.MongoDbSnap) {
+		if args.DataDir != snap.DefaultDataPath {
+			logger.Warningf("overwriting args.dataDir (set to %v) to %v", args.DataDir, snap.DefaultDataPath)
```

Comment:
Don't really care about if it's overwritten or not.
The important bit is - we're using the snap. So we just need a log message stating that the snap is used.
Also, it's not sufficient just to override data dir here. It needs to be set properly up front depending on if the snap is used, since other things like backups need to be able to access it.
In the code, we have a lot of this:

```
dbPath := filepath.Join(dataDir, "db")
```

You need to create a helper function and return the right thing depending on the feature flag